src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const newState = { ...state };
+
+  for (const { type, ...rest } of actions) {
```

Comment:
Consider using a switch statement instead of multiple if statements for handling the 'type' property. This will make the code cleaner and more maintainable.